server/src/main/java/org/opensearch/common/cache/policy/CachePolicyInfoWrapper.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.policy;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+
+import java.io.IOException;
+
+/**
+ * A class containing information needed for all CacheTierPolicy objects to decide whether to admit
+ * a given BytesReference. This spares us from having to create an entire short-lived QuerySearchResult object
+ * just to read a few values.
+ */
+public class CachePolicyInfoWrapper implements Writeable {
```

Comment:
The reason I was keeping the `CachedQueryResult` in the passed parser is because then you are defining a contract in terms of that class which any cache implementation can rely on. Just having a long makes it difficult to state what that long represents and how to use it.

Reply:
But if we do that, every time we inspect a BytesReference for policies, we have to create an entire short-lived CachedQueryResult object, which would be a lot larger than just a long or the CachePolicyInfoWrapper we originally had. We were thinking it would put less load on the GC this way.